### PR TITLE
groupByUnique

### PR DESCRIPTION
**Pull request by Adrian-Sanchez**
So i have a use-case...thought maybe you would too.
Basically, suppose your component:singleton has within something you query often. Some sort of lookup service.
ex:  var userQry = qb.from('users').get()
variables.users =  collect( userQry ).groupBy('id');
Currently, the groupBy will result ["1": [{id:"1",col2:"",col3:""...} ,"2":...]
so if you have to write your lookup service as getUsersById(1)
return variables.users[1][0]...
groupByUnique will result in ["1": {id:"1",col2:"",col3:""...} ,"2":{cols}...]
so anywhere you can now just return variables.users[1] 
Nested array gone because we expect a record as struct, not an array of structs

So, whether its groupByKey or groupByUnique or whatever you name you like...or if this isn't collection worthy...thought i'd share.

### CODE REVIEW

File: models/Collection.cfc [hunk 1]
```diff
@@ -409,6 +409,21 @@ component accessors="true" {
...
     }
+    public struct function groupByUnique( required string key, boolean forceLookup = false ) {
+        return this.reduce( function( acc, item ) {
+            if ( ( isObject( item ) && structKeyExists( item, "get#key#" ) ) || forceLookup ) {
+                var value = invoke( item, "get#key#" );
+            }
+            else {
+                var value = item[ key ];
+            }
+            if ( ! structKeyExists( acc, value ) ) {
```
Comment (elpete) [inline models/Collection.cfc]: I don't think you need these three lines if this is returning a struct.

**Comment by elpete** [comment]
It would be nice if this could be an alias for calling the original `groupBy` method with a boolean flag of `unique` or `keepOne`.

Also, just remember that this will keep only the last item if there are duplicates.

**Comment by elpete** [comment]
Oh, and also a test would be nice. :+1:

**Comment by Adrian-Sanchez** [comment]
So its your project and did it the way you would have it done.
Added unique argument
Defaults to false for compatibility.
But if it is true, will return unique structs and throw error if the key is not unique. This is to avoid any expectation that it's going to overwrite it.

File: models/Collection.cfc (hunk 1 above)
Comment (Adrian-Sanchez) [reply models/Collection.cfc]: By aliasing...n/a anymore

**Comment by elpete** [comment]
One last change - can you add this to the README?

**Comment by Adrian-Sanchez** [comment]
> One last change - can you add this to the README?

Done.

**Comment by elpete** [comment]
Thank you so much!

**Comment by Adrian-Sanchez** [comment]
No problem. Knowing its there, i don't have to dig around to include it in
the next project. win/win

**Pull request merged**